### DISCUSSION TITLE
Home Assistant proxy for Music Assistant

### DISCUSSION THREAD

**Discussion by Sarnog** [music-assistant/support#3102] [Feature requests and ideas]
I'd like to see a Home Assistant add-on with a proxy for Music Assistant. This way people can easely use the Music Assistant UI within Home Assistant, when Music Assistant is installed on a different machine and not as a add-on within Home Assistant. I hope you guys can help me out.

**Comment by OzGav** [comment]
What is wrong with using an iframe?

**Reply by Sarnog** [reply]
Http doesn't load on a https connection.

**Reply by OzGav** [reply]
I understand your request but you could setup your own reverse proxy

**Reply by Sarnog** [reply]
I did, but i don't want it exposed externally without some authentification. So i created a domain with authelia, but that doesn't worl with iframe. I also created an internal domain, but the link to that is only availabe via a VPN or on my local network. So that is the whole problem, why i ask to get a proxy.

**Reply by OzGav** [reply]
OK. Will see how many upvotes this gets

**Reply by Sarnog** [reply]
I already created one myself, feel free to use it:

https://github.com/Sarnog/Music-Assistant-Proxy

**Comment by darthrater78** [comment]
For what It's worth both Frigate and Z2m have Proxy add ons so they can be run externally and easily located in the HA sidebar. This would be great for MA. 

Edit: @Sarnog thanks for your proxy, I added via custom repos

**Reply by Sarnog** [reply]
Have fun with it... 😊

**Reply by Njur** [reply]
Is a revese proxy enough? I have configured on but if someone gets the address, it still has no login security?

**Reply by Sarnog** [reply]
> Is a revese proxy enough? I have configured on but if someone gets the address, it still has no login security?

It's an addon you can configure in Home Assistant if you're not using the Music Assistant addon in Home Assistant, but have Music Assistant on another machine for example. So it will connect via internal network (DHCP) instead of via the internet to your Music Assistant in your Home Assistant instance.

**Comment by MankiniChykan** [comment]
Up vote.

**Comment by MankiniChykan** [comment]
Will this work around this issue? Using Home Assistant Addon but Non-admin users aren’t allowed to subscribe to arbitrary events like mass_queue

[https://github.com/droans/mass_queue/issues/45](https://github.com/droans/mass_queue/issues/45)

**Reply by Sarnog** [reply]
I don't think so, but you can try it.

**Comment by OzGav** [comment]
I think this has been satisifed with the new Remote Access feature https://www.music-assistant.io/settings/remote-access/
